tests/common.go

```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
 	}
 	return clusterProviders
 }
+
+func IsPoolOffline(n node.Node) (bool, error) {
+	poolsStatus, err := Inst().V.GetNodePoolsStatus(n)
+	if err != nil {
+		return false, err
+	}
+
+	for _, v := range poolsStatus {
+		if v == "Offline" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func GetPoolUuidsWithStorageFull() ([]string, error) {
+	var poolUuids []string
+	pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+	if err != nil {
+		return nil, err
+	}
+	calculatePercentage := func(usedValue float64, totalValue float64) float64 {
+		percentage := (usedValue / totalValue) * 100
+		return float64(percentage)
+	}
+
+	for _, eachPool := range pools {
+		nodeDetail, err := GetNodeWithGivenPoolID(eachPool.Uuid)
+		if err != nil {
+			return nil, err
+		}
+
+		poolOfflineStatus, err := IsPoolOffline(*nodeDetail)
+		if err != nil {
+			return nil, err
+		}
+
+		if calculatePercentage(float64(eachPool.Used), float64(eachPool.TotalSize)) > 90.0 || poolOfflineStatus {
+			poolUuids = append(poolUuids, eachPool.GetUuid())
+		}
+	}
+
+	return poolUuids, nil
+}
+
+// GetVolumeConsumedSize returns size of the volume
+func GetVolumeConsumedSize(vol volume.Volume) (uint64, error) {
+	// Get Random Storage Node
+	cmd := fmt.Sprintf("pxctl v i %v -j | jq '.[].usage'", vol.ID)
+	output, err := runCmdGetOutput(cmd, node.GetStorageNodes()[0])
+	if err != nil {
+		return 0, err
+	}
+	output = strings.ReplaceAll(output, "\"", "")
+	output = strings.TrimSpace(output)
+
+	num, err := strconv.ParseUint(output, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return num, nil
+}
+
+// IsIORunningOnVolume return true if IO is running on the volume
+func IsIORunningOnVolume(vol volume.Volume) (bool, error) {
```

Comment:
can you reuse this IsIOsInProgressForTheVolume?

Reply:
fixed